opds_import.py

```diff
@@ -593,89 +547,84 @@ def data_detail_for_feedparser_entry(cls, entry, data_source):
             status_message = StatusMessage(status_code, message)
 
         if status_message and not status_message.success:
-            return identifier, None, None, status_message
+            return identifier, None, status_message
 
         # At this point we can assume that we successfully got some
         # metadata, and possibly a link to the actual book.
 
         # Note: will ignore bibframe_distribution from feed, and use data source passed into the 
         # importer as license_data_source, too.
 
-        title = entry.get('title', None)
-        if title == OPDSFeed.NO_TITLE:
-            title = None
-        subtitle = entry.get('schema_alternativeheadline', None)
-
-        def _datetime(key):
-            value = entry.get(key, None)
-            if not value:
-                return value
-            return datetime.datetime(*value[:6])
-
-        last_opds_update = _datetime('updated_parsed')
-        added_to_collection_time = _datetime('published_parsed')
-
-        publisher = entry.get('publisher', None)
-        if not publisher:
-            publisher = entry.get('dcterms_publisher', None)
-
-        language = entry.get('language', None)
-        if not language:
-            language = entry.get('dcterms_language', None)
-
-        links = []
-
-        def summary_to_linkdata(detail):
-            if not detail:
-                return None
-            if not 'value' in detail or not detail['value']:
-                return None
-
-            content = detail['value']
-            media_type = detail.get('type', 'text/plain')
-            return LinkData(
-                rel=Hyperlink.DESCRIPTION,
-                media_type=media_type,
-                content=content
-            )
+        try:
+            title = entry.get('title', None)
```

Comment:
Again, a big try/catch block is a great opportunity to create a testable helper method.